lua/nvim-tree/renderer.lua

```diff
@@ -44,7 +44,8 @@ if icon_state.show_folder_icon then
   end
   set_folder_hl = function(line, depth, icon_len, name_len, hl_group)
     table.insert(hl, {hl_group, line, depth+icon_len, depth+icon_len+name_len+get_trailing_length()})
-    table.insert(hl, {'NvimTreeFolderIcon', line, depth, depth+icon_len})
+    local hl_icon = vim.g.nvim_tree_highlight_opened_files and hl_group or 'NvimTreeFolderIcon'
```

Comment:
The config check here should be `(vim.g.nvim_tree_highlight_opened_files or 0) ~= 0`, because 0 is a truthy value in lua. And everywhere else we treat 0 as meaning disabled.